src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -638,13 +639,52 @@ bool mongoLdRegistrationsGet
     }
   }
 
+  if (ciP->uriParam["type"] != "")
+  {
+    char*                       typeList = (char*) ciP->uriParam["type"].c_str();
+    std::vector<std::string>    typeVec;
+    int                         types;
+    mongo::BSONObjBuilder       bsonInExpression;
+    mongo::BSONArrayBuilder     bsonArray;
+    char*                       details;
+    char                        typeExpanded[256];
+
+    types = stringSplit(typeList, ',', typeVec);
+
+    if (types > 0)
+    {
+      for (int ix = 0; ix < types; ix++){
+        char type[typeVec[ix].size() + 1];
+        strcpy(type, typeVec[ix].c_str());
+
+        if (((strncmp(type, "http://", 7) == 0) || (strncmp(type, "https://", 8) == 0)) && (urlCheck(type, &details) == true))
+        {
+          // No expansion desired, the type is already a FQN
+          strncpy(typeExpanded, type, sizeof(typeExpanded));
+        }
+        else if (orionldUriExpand(orionldState.contextP, type, typeExpanded, sizeof(typeExpanded), &details) == false)
+        {
+          orionldErrorResponseCreate(OrionldBadRequestData, "Error during URI expansion of entity type", details, OrionldDetailsString);
+          return false;
+        }
+
+        typeVec[ix] = typeExpanded;
```

Comment:
This line will also disappear (`typeVec[ix] = typeExpanded`), and ...

Reply:
Right!

Reply:
Fixed in 818e634